base/math.jl

```diff
@@ -579,12 +579,18 @@ function hypot(x::T,y::T) where T<:AbstractFloat
         scale = one(scale)
     end
     h = sqrt(muladd(ax,ax,ay*ay))
-    if h <= 2*ay
-        delta = h-ay
-        h -= muladd(delta,delta-2*(ax-ay),ax*(2*delta - ax))/(2*h)
+    if Base.Math.FMA_NATIVE
+        hsquared = h*h
```

Comment:
Maybe add a comment here explaining the reason for this branch.

`# correctly rounded variant when hardware fma is available, preserves performance`   or similar...

Reply:
Excellent suggestion. I will make an appropriate change